sdk/Sdk/FunctionMetadataGenerator.cs

```diff
@@ -378,14 +372,223 @@ private static ExpandoObject BuildBindingMetadataFromAttribute(CustomAttribute a
+            }
+
+            TypeDefinition baseType = definition.BaseType.Resolve();
+            return IsDerivedFrom(baseType, interfaceFullName);
+        }
+
+        private static string? ResolveIEnumerableOfTType(TypeReference type, Dictionary<string, string> foundMapping)
+        {
+            // Base case: 
+            // We are at IEnumerable<T> and want to return the most recent resolution of T
+            // (Most recent is relative to IEnumerable<T>)
+            if (string.Equals(type.FullName, Constants.IEnumerableOfT, StringComparison.Ordinal))
+            {
+                if (foundMapping.TryGetValue(Constants.GenericIEnumerableArgumentName, out string typeName))
+                {
+                    return typeName;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+
+            TypeDefinition definition = type.Resolve();
+            if (definition.HasGenericParameters && type is GenericInstanceType genericType)
+            {
+                for (int i = 0; i < genericType.GenericArguments.Count(); i++)
+                {
+                    string name = genericType.GenericArguments.ElementAt(i).FullName;
+                    string resolvedName = definition.GenericParameters.ElementAt(i).FullName;
+
+                    if (foundMapping.TryGetValue(name, out string firstType))
+                    {
+                        foundMapping.Remove(name);
+                        foundMapping.Add(resolvedName, firstType);
+                    }
+                    else
+                    {
+                        foundMapping.Add(resolvedName, name);
+                    }
+                }
+
+            }
+
+            return definition.Interfaces
+                    .Select(i => ResolveIEnumerableOfTType(i.InterfaceType, foundMapping))
+                    .Where(name => name is not null)
+                    .FirstOrDefault()
+                ?? definition.NestedTypes
+                    .Select(t => ResolveIEnumerableOfTType(t, foundMapping))
+                    .Where(name => name is not null)
+                    .FirstOrDefault()
+                ?? ResolveIEnumerableOfTType(definition.BaseType, foundMapping);
+        }
+
+        private static DataType GetDataTypeFromType(string fullName)
+        {
+            if (IsStringType(fullName))
+            {
+                return DataType.String;
+            }
+            else if (IsBinaryType(fullName))
+            {
+                return DataType.Binary;
+            }
+
+            return DataType.Undefined;
+        }
+
+        private static bool IsStringType(string fullName)
+        {
+            return string.Equals(fullName, Constants.StringType, StringComparison.Ordinal);
+        }
+
+        private static bool IsBinaryType(string fullName)
+        {
+            return string.Equals(fullName, Constants.ByteArrayType, StringComparison.Ordinal);
```

Comment:
also check if type is IReadOnlyMemory<T>

Reply:
Note: type is ReadOnlyMemory<T> 
proposal for IReadOnlyMemory<T>: https://github.com/dotnet/runtime/issues/32706